src/index.js

```diff
@@ -184,7 +195,19 @@ exports.createFromBytes = (buf) => {
 }
 
 exports.createFromB58String = (str) => {
-  return new PeerIdWithIs(mh.fromB58String(str))
+  return exports.createFromCID(str) // B58String is CIDv0
+}
+
+exports.createFromCID = (cid) => {
+  if (typeof cid === 'string' || Buffer.isBuffer(cid)) {
```

Comment:
These checks are done in the CID constructor, we shouldn't repeat them here!

Reply:
The reason for this was to provide a meaningful error `Supplied cid value is neither String|CID|Buffer`. 

If we delegate validation to CID constructor, it will spit out unhelpful  [`Invalid version, must be a number equal to 1 or 0`](https://github.com/multiformats/js-cid/blob/6c215bdc1a2440d5bccc0801327c1ef123e93a10/src/cid-util.js#L20) in most of error cases.